### PR TITLE
Use PSRule for Security testing and re-enable manual testing

### PR DESCRIPTION
**Pull request by tonybaloney**
## Purpose

This uses a different rule for security testing that reduces false-positives and validates only what is deployed.

### CODE REVIEW

File: .github/workflows/azure-dev-validation.yaml [hunk 1]
```diff
@@ -8,6 +8,7 @@ on:
     branches: [ aca-openai-agent ]
```
Comment (marlenezw) [inline .github/workflows/azure-dev-validation.yaml:8]:
```suggestion
    branches: [ main ]
```

File: infra/main.test.bicep [hunk 2]
```diff
@@ -0,0 +1,17 @@
+// This file is for doing static analysis and contains sensible defaults
+// for PSRule to minimise false-positives and provide the best results.
+
+// This file is not intended to be used as a runtime configuration file.
+
+targetScope = 'subscription'
+
+param environmentName string = 'testing'
+param location string = 'swedencentral'
```
Comment (marlenezw) [inline infra/main.test.bicep:9]: Should we use Sweden central here or East US 2 as well?

**Comment by marlenezw** [comment]
This looks good to me too! Just realised I need to update the workflow branch and confirming if the region. I think Sweden Central doesn't have semantic search functionality that is needed for this project

File: infra/main.test.bicep (hunk 2 above)
Comment (tonybaloney) [reply infra/main.test.bicep:9]: As long as it's one of the allowed regions,it doesn't matter

**Comment by tonybaloney** [comment]
> This looks good to me too! Just realised I need to update the workflow branch and confirming if the region. I think Sweden Central doesn't have semantic search functionality that is needed for this project

That should probably be fixed separately then because it's one of the allowed regions and the most popular for AI deployments outside of the US

**Comment by marlenezw** [comment]
Thanks for the PR, I'll document the issue!!!

**Pull request merged**